psygnal/_signal.py

```diff
@@ -912,3 +919,40 @@ def _is_subclass(left: AnyType, right: type) -> bool:
         if origin is Union:
             return any(issubclass(i, right) for i in get_args(left))
     return issubclass(left, right)
+
+
+def partial_weakref(partial_fun: partial) -> Callable:
+    obj, name = _get_proper_name(partial_fun.func)
+    args_ = partial_fun.args
+    kwargs_ = partial_fun.keywords
+
+    def wrap(*args, **kwargs):  # type: ignore
+        ob = obj()
+        if ob is None:
+            return
+        getattr(ob, name)(*args_, *args, **kwargs_, **kwargs)
+
+    wrap.__ref__ = obj  # type: ignore
+
+    del partial_fun
+    return wrap
+
+
+def _get_proper_name(callback: Callable) -> Tuple[weakref.ref, str]:
+    assert inspect.ismethod(callback)
+    obj = callback.__self__  # type: ignore
```

Comment:
There is any option to annotate the function as a method (to remove this ignore)?

Reply:
we could maybe use `types.MethodType` ... and instead of `inspect.ismethod` (Which apparently doesn't do the required type narrowing) we can use `isinstance(obj, MethodType)` ... then make the argument `callback: MethodType` instead of `Callable`?